### DISCUSSION TITLE
"// +build wireinject" cause gopls error

### DISCUSSION THREAD

**Discussion by ymchun** [google/wire#283] [Q&A]
### Describe the bug

If we have a go file that call `wire.Build`, we put a build constraint to the start of the file. The `gopls` will prompt the following message saying that no packages found.

```
No packages found for open file /some/path/wire.go: <nil>.
If this file contains build tags, try adding "-tags=<build tag>" to your gopls "buildFlag" configuration (see (https://github.com/golang/tools/blob/master/gopls/doc/settings.md#buildflags-string).
Otherwise, see the troubleshooting guidelines for help investigating (https://github.com/golang/tools/blob/master/gopls/doc/troubleshooting.md).
```

If we remove the empty line between the build constraint and the `package`, it behave like the issue described here https://github.com/google/wire/issues/117

### To Reproduce

```go
// +build wireinject

package cmd
```

## Expected behavior

No warnings.

### Version

```
go version go1.16 darwin/amd64
```

### Additional context

My vscode settings

```json
{
  "go.lintOnSave": "workspace",
  "go.lintTool": "golangci-lint",
  "go.useLanguageServer": true,
  "go.formatTool": "goimports",
  "[go]": {
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
      "source.organizeImports": true
    },
    "editor.snippetSuggestions": "none"
  },
  "[go.mod]": {
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
      "source.organizeImports": true
    }
  },
  "gopls": {
    "usePlaceholders": true,
    "staticcheck": false
  },
}
```

**Comment by zombiezen** [comment]
See some of the workarounds in golang/go#29202. I think this one works, but untested:

```json
  "gopls": {
    "buildFlags": ["-tags=wireinject"]
  },
```

**Reply by bagualing** [reply]
I got the same error even if i added "buildFlags": ["-tags=wireinject"].

**Comment by imxw** [comment]
> I got the same error even if i added "buildFlags": ["-tags=wireinject"].

```
        "build.standaloneTags": [
            "ignore",
            "wireinject",
            "!wireinject"
          ]
```
It works for me.

**Reply by sm-tunglt** [reply]
I also setup like you. It seem like generated code is not included when build, and got this error message. Can you have any suggestion?

`panic: implementation not generated, run wire`
